lib/inspector/promises.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const inspector = require('inspector');
+const { promisify } = require('util');
+class Session extends inspector.Session {
+  #post = promisify(super.post.bind(this));
+  /**
+   * Posts a message to the inspector back-end.
+   * @param {string} method
+   * @param {Record<unknown, unknown>} [params]
+   * @returns {Promise}
+   */
+  async post(method, params) {
+    return this.#post(method, params);
+  }
+}
```

Comment:
```suggestion

class Session extends inspector.Session {}

Session.prototype.post = promisify(inspector.Session.prototype.post);
```

Reply:
Why is this better? The private method is robust; this can be mutated.

Reply:
Both versions can be mutated, it’s better because it’s less code.

Reply:
We should be optimizing for less observable/mutable runtime behavior not less code.

Reply:
Put another way, more declarative code is far better than less imperative code.

Reply:
I suppose it's a point of vue, one could argue we should write code that produces the least amount of JS objects to save some memory. Anyway, @EricWendel pick the version you prefer, my suggestion is not blocking (although, if you're not taking my suggestion, please use `FunctionPrototypeBind` instead of `.bind` line 6).

Reply:
Thanks @aduh95 and @ljharb just changed it!